python/lsst/ts/externalscripts/base_make_calibrations.py

```diff
@@ -296,17 +311,47 @@ async def configure(self, config):
         config: `types.SimpleNamespace`
             Configuration data. See `get_schema` for information about data
             structure.
+
+        Raises
+        ------
+            RuntimeError : Incorrect instrument name.
         """
         # Log information about the configuration
 
         self.log.debug(
             f"n_bias: {config.n_bias}, detectors: {config.detectors}, "
             f"n_dark: {config.n_dark}, "
             f"n_flat: {config.n_flat}, "
-            f"instrument: {self.instrument_name} "
+            f"instrument: {self.instrument_name}, "
+            f"script_mode: {config.script_mode}, "
+            f"generate_calibrations: {config.generate_calibrations}, "
+            f"do_verify: {config.do_verify}, "
+            f"do_defects: {config.do_defects}, "
+            f"do_ptc: {config.do_ptc}, "
+            f"do_gain_from_flat_pairs: {config.do_gain_from_flat_pairs} "
         )
 
+        if self.instrument_name == "LATISS":
```

Comment:
I don't think this if/else statement should be here. It is kind of doing an inline specialization in the base class.

I don't think LATISS need to deal with detectors at all.

If this is something you need to handle on the parent class, then make an abstract method that is called to handle it and implement if on the child classes.

Reply:
I do need the detectors for the pipetasks. I'll add an abstract class as you suggest.